tests/unit_tests/grid/test_xyz.py

```diff
@@ -0,0 +1,656 @@
+                  xy_units = 'm',
+                  z_units = 'm',
+                  axis_order = 'easting northing',
+                  epsg_code = '32630')
+    crs.create_xml()
+
+    global_xy_units = 'ly'
+    global_z_units = 'ly'
+
+    a = np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]])
+
+    # Act & Asert
+    with pytest.raises(InvalidUnitError):
+        basic_regular_grid.local_to_global_crs(a,
+                                               crs_uuid = crs.uuid,
+                                               global_xy_units = global_xy_units,
+                                               global_z_units = global_z_units)
+
+
+def test_local_to_global_crs_global_z_increasing_downward(basic_regular_grid):
+    # Arrange
+    x_offset = 0.0
+    y_offset = 0.0
+    z_offset = 0.0
+    rotation = 0.0
+    rotation_units = 'dega'
+
+    crs = rqc.Crs(basic_regular_grid.model,
+                  x_offset = x_offset,
+                  y_offset = y_offset,
+                  z_offset = z_offset,
+                  z_inc_down = True,
+                  rotation = rotation,
+                  rotation_units = rotation_units,
+                  xy_units = 'm',
+                  z_units = 'm',
+                  axis_order = 'easting northing',
+                  epsg_code = '32630')
+    crs.create_xml()
+
+    global_xy_units = 'ft'
+    global_z_units = 'ft'
+    global_z_increasing_downward = False
+
+    a = np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]])
+    a_expected = np.array([[1.0, 1.0, -1.0], [1.0, 1.0, -1.0], [1.0, 1.0, -1.0]]) * METRES_TO_FEET
+
+    # Act
+    a_transformed = basic_regular_grid.local_to_global_crs(a,
+                                                           crs_uuid = crs.uuid,
+                                                           global_xy_units = global_xy_units,
+                                                           global_z_units = global_z_units,
+                                                           global_z_increasing_downward = global_z_increasing_downward)
+
+    # Assert
+    np.testing.assert_array_almost_equal(a_transformed, a_expected)
+
+
+def test_local_to_global_crs_degree_rotation(basic_regular_grid):
+    # Arrange
+    x_offset = 0.0
+    y_offset = 0.0
+    z_offset = 0.0
+    rotation = 60
+    rotation_units = 'dega'
+
+    crs = rqc.Crs(basic_regular_grid.model,
+                  x_offset = x_offset,
+                  y_offset = y_offset,
+                  z_offset = z_offset,
+                  z_inc_down = True,
+                  rotation = rotation,
+                  rotation_units = rotation_units,
+                  xy_units = 'm',
+                  z_units = 'm',
+                  axis_order = 'easting northing',
+                  epsg_code = '32630')
+    crs.create_xml()
+
+    global_xy_units = 'm'
+    global_z_units = 'm'
+
+    a = np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]])
```

Comment:
could be good to at least use different points on the diagonal here, eg. [2.0, 2.0, 2.0] and [-1.0, -1.0, -1.0]